addon/-private/system/model/record-data.ts

```diff
@@ -5,10 +5,33 @@ import { assert, warn, inspect } from '@ember/debug';
 import { run } from '@ember/runloop';
 import Relationships from '../relationships/state/create';
 import coerceId from '../coerce-id';
+import BelongsToRelationship from '../relationships/state/belongs-to';
+import ManyRelationship from '../relationships/state/has-many';
+import Relationship from '../relationships/state/relationship';
+import RecordData, { ChangedAttributesHash } from '../../ts-interfaces/record-data'
+import { JsonApiResource, JsonApiResourceIdentity, JsonApiBelongsToRelationship, JsonApiHasManyRelationship, AttributesHash } from "../../ts-interfaces/record-data-json-api";
+import { RelationshipRecordData } from '../../ts-interfaces/relationship-record-data';
+import { RecordDataStoreWrapper } from '../../ts-interfaces/record-data-store-wrapper';
 
 let nextBfsId = 1;
-export default class RecordData {
-  constructor(modelName, id, clientId, storeWrapper, store) {
+
+export default class RecordDataDefault implements RelationshipRecordData {
+  store: any;
+  modelName: string;
+  __relationships: Relationships | null;
+  __implicitRelationships:{ [key: string]: Relationship } | null;
+  clientId: string;
+  id: string | null;
+  storeWrapper: RecordDataStoreWrapper;
+  isDestroyed: boolean;
+  _isNew: boolean;
+  _bfsId: number;
+  __attributes: any;
+  __inFlightAttributes: any;
+  __data: any;
+  _scheduledDestroy: any;
+
+  constructor(modelName: string, id: string, clientId: string, storeWrapper: RecordDataStoreWrapper, store:any) {
```

Comment:
I am surprised that this did not yell at us from within `InternalModel` but `id` here is `null` in the new record case fairly often.

Reply:
It's called from inside store, which isn't in ts yet

Reply:
Fixed